distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/TestFederation.java

```diff
@@ -44,56 +46,83 @@
+    private static final String CORBA_DEFAULT_PORT_PROPERTY =
+            "org.codice.alliance.corba_default_port";
 
-    private static final String HTTP_NSILI_SOURCE_ID = "httpNsiliSource";
+    private static final DynamicPort CORBA_DEFAULT_PORT = new DynamicPort(
+            CORBA_DEFAULT_PORT_PROPERTY,
+            6);
 
-    private static final String FTP_NSILI_SOURCE_ID = "ftpNsiliSource";
+    private static final DynamicPort HTTP_WEB_PORT = new DynamicPort(
+            "org.codice.alliance.corba_web_port",
+            7);
+
+    private static final DynamicPort FTP_WEB_PORT = new DynamicPort(
+            "org.codice.alliance.corba_ftp_web_port",
+            8);
+
+    private static final DynamicPort CSW_STUP_SERVER_PORT = new DynamicPort(
+            "org.codice.alliance.csw_stub_server_port",
+            9);
+
+    private static final DynamicPort CORBA_PORT = new DynamicPort("org.codice.alliance.corba_port",
+            10);
+
+    private static final DynamicUrl CSW_STUB_SERVER_URL = new DynamicUrl(INSECURE_ROOT,
+            CSW_STUP_SERVER_PORT,
+            "/services/csw");
+
+    private FederatedMgmpMockServer cswServer;
+
+    private Thread mockServerThread;
 
     @Override
     protected Option[] configureDistribution() {
-        return options(
-                karafDistributionConfiguration(
-                        maven().groupId("org.codice.alliance.distribution").artifactId("alliance")
-                                .type("zip").versionAsInProject().getURL(),
-                        "alliance", KARAF_VERSION).unpackDirectory(new File("target/exam"))
-                        .useDeployFolder(false));
+        return options(karafDistributionConfiguration(maven().groupId(
+                "org.codice.alliance.distribution")
+                .artifactId("alliance")
+                .type("zip")
+                .versionAsInProject()
+                .getURL(), "alliance", KARAF_VERSION).unpackDirectory(new File("target/exam"))
+                .useDeployFolder(false));
     }
 
     @Override
     protected Option[] configureCustom() {
-        return options(
-                wrappedBundle(mavenBundle("ddf.test.itests", "test-itests-ddf").classifier("tests")
-                        .versionAsInProject()).bundleSymbolicName("test-itests-ddf")
-                        .exports("ddf.test.itests.*"),
-                wrappedBundle(mavenBundle().groupId("org.codice.alliance.distribution")
-                        .artifactId("sample-nsili-server").versionAsInProject()),
-                keepRuntimeFolder());
+        return options(wrappedBundle(mavenBundle("ddf.test.itests", "test-itests-ddf").classifier(
+                "tests")
+                .versionAsInProject()).bundleSymbolicName("test-itests-ddf")
+                .exports("ddf.test.itests.*"), wrappedBundle(mavenBundle().groupId(
+                "org.codice.alliance.distribution")
+                .artifactId("sample-nsili-server")
```

Comment:
We are actually extending the `FederatedCswMockServer` from DDF and using that in the integration tests.  The sample-mgmp-server is intended to be used for development purposes.